pkg/engine/datasource/graphql_datasource/graphql_subscription_client.go

```diff
@@ -132,19 +132,30 @@ func (c *WebSocketGraphQLSubscriptionClient) Subscribe(ctx context.Context, opti
 	if err != nil {
 		return err
 	}
-	msgType, connectionAckMsg, err := conn.Read(ctx)
-	if err != nil {
-		return err
-	}
-	if msgType != websocket.MessageText {
-		return fmt.Errorf("unexpected msg type")
-	}
-	connectionAck, err := jsonparser.GetString(connectionAckMsg, "type")
-	if err != nil {
-		return err
-	}
-	if connectionAck != "connection_ack" {
-		return fmt.Errorf("expected connection_ack, got: %s", connectionAck)
+	// wait for ack and/or ka
+ackWait:
+	for {
+		msgType, msg, err := conn.Read(ctx)
+		if err != nil {
+			return err
+		}
+		if msgType != websocket.MessageText {
+			return fmt.Errorf("unexpected message type")
+		}
+
+		respType, err := jsonparser.GetString(msg, "type")
+		if err != nil {
+			return err
+		}
+
+		switch respType {
+		case "ka":
+			continue
+		case "connection_ack":
+			break ackWait
```

Comment:
This is a potential infinite loop, should there be a timeout here, and what do we want it to be (couldn't see it in any spec)

Reply:
yeah, this makes sense. We can start with 30s for example and tweak it down if needed